src/main/java/com/aws/greengrass/clientdevices/auth/iot/Certificate.java

```diff
@@ -116,6 +122,20 @@ public Map<String, DeviceAttribute> getDeviceAttributes() {
         return Collections.singletonMap("CertificateId", new StringLiteralAttribute(getCertificateId()));
     }
 
+    /**
+     * Updates the duration for which a certificate metadata can be trusted.
+     *
+     * @param newTrustDuration desired trust duration in hours
+     */
+    public static void updateMetadataTrustDurationHours(int newTrustDuration) {
+        metadataTrustDurationHours.set(newTrustDuration);
+    }
+
+    private boolean isStatusTrusted() {
+        Instant validTill = statusLastUpdated.plus(metadataTrustDurationHours.get(), ChronoUnit.HOURS);
+        return validTill.isAfter(Instant.now());
```

Comment:
if we can inject a `Clock` that'd be great for testing. `clock.instant()`

Reply:
That would be awesome. We can also mock the clock though and make `Instant.now` return any value we want

Reply:
agreed, let me have that in a separate PR.